src/main/groovy/pl/allegro/tech/build/axion/release/domain/VersionSorter.groovy

```diff
@@ -24,20 +24,21 @@ class VersionSorter {
                                   boolean ignoreNextVersionTags,
                                   boolean forceSnapshot,
                                   Pattern nextVersionTagPattern,
-                                  VersionFactory versionFactory) {
+                                  VersionFactory versionFactory,
+                                  String latestRelevantPosition) {
         Set<Version> versions = []
         Map<Version, Boolean> isVersionNextVersion = [:]
         Map<Version, TagsOnCommit> versionToCommit = new LinkedHashMap<>()
 
         for (TagsOnCommit tagsEntry : taggedCommits) {
             List<String> tags = tagsEntry.tags
 
-            // next version should be igored when tag is on head
+            // next version should be igored when tag is on revision of the latest change in the relevant folder
             // and there are other, normal tags on it
             // because when on single commit on head - normal ones have precedence
             // however, we should take into account next version
             // in case of forced snapshot
-            boolean ignoreNextVersionOnHead = tagsEntry.isHead &&
```

Comment:
I don't really understand the meaning of this line. If this variable wasn't there, VersionSorter wouldn't need to be changed at all.

Reply:
This fixes one of the corner cases when multiple tags are on the same tag. Imagine commit with tags:

```
abcf124 -> Work in progress on 2.0.0
abcf123 -> release-1.0.0 release-2.0.0-alpha // alpha is nextVersion tag
```

In following situations expected outcomes are:

```
git checkout abcf124
./gradlew cV
2.0.0-SNAPSHOT # I'm working on next version

git checkout abcf123
./gradlew cV
1.0.0 # I'm on released commit now
```

Thus check for `isHead` is necessary for version sorting. I don't know if this is necessary in your case, because it serves different purpose than establishing if I'm on SNAPSHOT or not.